File: CoursesOnDemand/ClientApp/src/app/_models/student.ts
import { CreditCard } from './creditcard';
import { Course } from './course';

export class Student {
  id: number;
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  //User data attributes
  password: string;
  token: string;
  Course: Course;
  CreditCard: CreditCard
}

File: CoursesOnDemand.Models/Student.cs
﻿using System;
using System.Collections.Generic;
using System.Text;

namespace CoursesOnDemand.Domain
{
    public class Student : User
    {
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public String Email { get; set; }
        public String Phone { get; set; }
        public CreditCard CreditCard { get; set; }
        public Course Course { get; set; }
        public IList<Payment> Payments { get; set; }        
    }
}
